dev/app.py

```diff
@@ -23,11 +28,84 @@ def set_title(self, title, font_size=None):
     def load_screen(self):
         lv.scr_load(self.scr)
 
-    def add_item(self, name, item, selectable=False):                
+    def add_item(self, name, item, selectable=False):
         self.items[name] = item
         self.item_ids[id(item.lv_obj)] = item
         if selectable:
             lv.group_add_obj(self.group, item.lv_obj)
 
     def get_cont(self):
         return self.cont
+
+
+class GameApp():
+
+    def __init__(self, name, display, th=MATERIAL_THEME, roll_over=False, border=False, kill=False):
+        self.theme = th
+        lv.theme_set_current(self.theme)
+        self.scr = lv.obj()
+        self.name = name
+        self.disp = display
+        self.game = g.Game(32, 32)
+        self.sprites = {}
+        self.roll_over = roll_over
+        self.border = border
+        self.kill = kill
+        self.load_screen()
+        # self.draw_screen()
+
+    def load_screen(self):
+        lv.scr_load(self.scr)
+
+    def draw_screen(self):
+        # 160 x 128 screen
+        # 5 x 4 for each element in grid
+        for x in range(self.game.x):
+            for y in range(self.game.y):
+                if self.game.present_at(x, y) != "ball":
+                    Rectangle(self.scr, x*5, y*4, (x*5)+5, (y*4)+4)
+
+    def draw_initial_sprite(self, sprite):
+        x = sprite.x
+        y = sprite.y
+        width = sprite.width
+        height = sprite.height
+        if sprite.type == "BALL":
+            sprite.set_icon(Ball(self.scr, width*5, height*4, x*5, y*4))
+        elif sprite.type == "PADDLE":
+            sprite.set_icon(PongBoard(self.scr, width*5, height*4, x*5, y*4))
+        else:
+            sprite.set_icon(Wall(self.scr, width, height, x, y))
```

Comment:
Would it be better to have this as another `elif` and then having a catch-all else, for any erroneous sprite types?
That would hopefully make it slightly more future-proof and easier to spot mistakes.

e.g.
```python
        if sprite.type == "BALL":
            sprite.set_icon(Ball(self.scr, width*5, height*4, x*5, y*4))
        elif sprite.type == "PADDLE":
            sprite.set_icon(PongBoard(self.scr, width*5, height*4, x*5, y*4))
        elif sprite.type == "WALL":
            sprite.set_icon(Wall(self.scr, width, height, x, y))
        else:
            pass # or print debug message, or similar
```

Reply:
Ah yeah I had initially done it like this so that any undefined sprite is drawn as rectangles but forgot to update after adding Wall(). Wall() can actually just be used to draw a grid of rectangles so maybe it should be renamed and used again here? I'll try a fix and see what you think